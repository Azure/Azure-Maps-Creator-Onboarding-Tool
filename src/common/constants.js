export const PATHS = {
  INDEX: '/',
  VIEW_CONVERSIONS: '/view-conversions',
  CREATE_UPLOAD: '/create-upload',
  PROCESSING: '/processing',
  CREATE_GEOREFERENCE: '/create-georeference',
  REVIEW_CREATE: '/review-create',
  CONVERSION: '/review-create/conversion',
  IMDF_CONVERSION: '/review-create/imdf-conversion',
  PAST_CONVERSION: '/past-conversion',
  CONVERSIONS: '/conversions',
  LAYERS: '/layers',
  LEVELS: '/levels',
  INVALID_PATH: '/*',
};

export const ROUTE_NAME_BY_PATH = {
  [PATHS.INDEX]: 'home',
  [PATHS.CREATE_UPLOAD]: 'Create',
  [PATHS.VIEW_CONVERSIONS]: 'View',
  [PATHS.PROCESSING]: 'processing',
  [PATHS.CREATE_GEOREFERENCE]: 'prepare.drawing.package',
  [PATHS.LAYERS]: 'prepare.drawing.package',
  [PATHS.LEVELS]: 'prepare.drawing.package',
  [PATHS.INVALID_PATH]: 'redirect',
  [PATHS.REVIEW_CREATE]: 'prepare.drawing.package',
  [PATHS.CONVERSION]: 'create.indoor.map',
  [PATHS.IMDF_CONVERSION]: 'convert.indoor.map',
  [PATHS.CONVERSIONS]: 'All conversions',
};

export const PLACES_ROUTE_NAME_BY_PATH = {
  [PATHS.CREATE_GEOREFERENCE]: 'prepare.building.configuration',
  [PATHS.LAYERS]: 'prepare.building.configuration',
  [PATHS.LEVELS]: 'prepare.building.configuration',
  [PATHS.REVIEW_CREATE]: 'prepare.building.configuration',
};

export const TRUNCATE_FRACTION_DIGITS = 8;

export const HTTP_STATUS_CODE = {
  ACCEPTED: 202,
  OK: 200,
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  FORBIDDEN: 403,
  TOO_MANY_REQUESTS: 429,
};

export const isStagingEnv = process.env.REACT_APP_STAGING_ENV === 'true';

export const PLACES_PREVIEW = {
  DEFAULT_LANGUAGE: 'en',
  VERSION: 'places-1.0',
  DESCRIPTION: 'Converted by AzMapsCreatorOnboardingTool',
  STORAGE_RETENTION: 'day',
  SEARCH_PARAMETER: 'placespreview',
};
