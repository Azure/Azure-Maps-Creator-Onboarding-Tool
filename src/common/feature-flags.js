const flags = {
  onboardingEnabled: false,
};

export default flags;