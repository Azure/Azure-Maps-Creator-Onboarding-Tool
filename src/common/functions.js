/**
 * Given a single path name, returns a list of all path names from root to the given path name.
 * Ex. Given '/create/hello', returns ['/', '/create', '/create/hello']
 * @param {String} pathName the path name of the URL
 * @returns {String[]} an array of path names
 */
export const getSplitPaths = (pathName) => {
  if (typeof pathName !== 'string' || pathName === '') { return []; }
  if (pathName === '/') { return ['/']; }

  const pathNameLowerCase = pathName.toLowerCase();
  const splits = pathNameLowerCase.split('/');
  const splitPaths = [];
  for (let i = splits.length - 1; i > 0; i--) {
    let stringBuilder = '';
    for (let j = 1; j <= i; j++) {
      stringBuilder += '/' + splits[j];
    }
    splitPaths.push(stringBuilder);
  }
  splitPaths.push('/');
  splitPaths.reverse();

  return splitPaths;
};