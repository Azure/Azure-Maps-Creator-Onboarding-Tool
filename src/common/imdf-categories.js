export const imdfCategories = [
  'auditorium',
  'brick',
  'classroom',
  'column',
  'concrete',
  'conferenceroom',
  'drywall',
  'elevator',
  'escalator',
  'fieldofplay',
  'firstaid',
  'fitnessroom',
  'foodservice',
  'footbridge',
  'glass',
  'huddleroom',
  'kitchen',
  'laboratory',
  'library',
  'lobby',
  'lounge',
  'mailroom',
  'mothersroom',
  'movietheater',
  'movingwalkway',
  'nonpublic',
  'office',
  'opentobelow',
  'parking',
  'phoneroom',
  'platform',
  'privatelounge',
  'ramp',
  'recreation',
  'restroom',
  'restroom.family',
  'restroom.female',
  'restroom.female.wheelchair',
  'restroom.male',
  'restroom.male.wheelchair',
  'restroom.transgender',
  'restroom.transgender.wheelchair',
  'restroom.unisex',
  'restroom.unisex.wheelchair',
  'restroom.wheelchair',
  'road',
  'room',
  'serverroom',
  'shower',
  'smokingarea',
  'stairs',
  'steps',
  'storage',
  'structure',
  'terrace',
  'theater',
  'unenclosedarea',
  'unspecified',
  'vegetation',
  'waitingroom',
  'walkway',
  'walkway.island',
  'wood',
];
