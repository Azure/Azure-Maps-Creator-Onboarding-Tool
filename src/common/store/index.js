export { conversionSteps, useConversionStore } from './conversion.store';
export { useDissolvedExterior, useGeometryStore } from './geometry.store';
export { useLayersStore } from './layers.store';
export { useLevelsStore } from './levels.store';
export {
  progressBarStepsByKey,
  useCompletedSteps,
  useProgressBarSteps,
  useProgressBarStore,
} from './progress-bar-steps';
export { resetStores } from './reset';
export { LRO_STATUS, useResponseStore } from './response.store';
export { useReviewManifestJson, useReviewManifestStore } from './review-manifest.store';
export { getDomain, useUserStore } from './user.store';
