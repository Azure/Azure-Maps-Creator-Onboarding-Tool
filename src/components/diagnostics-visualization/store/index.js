import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from '../reducers';

// Middlewares go here
const middleware = {};

const configureStore = (initialState) => createStore(
  rootReducer,
  initialState,
  // Redux Dev Tools Middleware
  composeWithDevTools(applyMiddleware(...middleware)),
);

// pass an optional param to rehydrate state on app start
const store = configureStore();

// export store singleton instance
export default store;
