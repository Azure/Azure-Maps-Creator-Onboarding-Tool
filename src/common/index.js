import { PATHS, ROUTE_NAME_BY_PATH } from './constants';

export { PATHS, ROUTE_NAME_BY_PATH };