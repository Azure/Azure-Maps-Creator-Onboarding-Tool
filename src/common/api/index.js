// The following API is designed to be used only by the onboarding tool and is not supported for any other use.
import { getEnvs } from 'common/functions';
import { useUserStore } from '../store/user.store';

export const uploadFile = (file) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  const url = `${getEnvs()[geography].URL}/manifest?api-version=2.0&subscription-key=${subscriptionKey}`;

  return fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/zip',
    },
    body: file,
  });
};

export const fetchFromLocation = (location) => {
  const { subscriptionKey } = useUserStore.getState();
  return fetch(`${location}&subscription-key=${subscriptionKey}`, {
    method: 'GET',
  });
};

export const deleteFromLocation = (fetchUrl) => {
  const { subscriptionKey } = useUserStore.getState();
  fetch(`${fetchUrl}&subscription-key=${subscriptionKey}`, {
    method: 'DELETE',
  });
};

export const fetchAddress = (address) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  return fetch(`${getEnvs()[geography].URL}/search/address/json?subscription-key=${subscriptionKey}&api-version=1.0&query=${address}&limit=1`)
    .then((res) => res.json());
};