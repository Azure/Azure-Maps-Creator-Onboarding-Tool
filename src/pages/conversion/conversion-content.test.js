import { screen, render, fireEvent } from '@testing-library/react';

import ConversionContent from './conversion-content';
import { useConversionStore } from 'common/store';

describe('conversion content', () => {
  beforeEach(() => {
    useConversionStore.setState({
      selectedStep: 0,
      conversionOperationLog: null,
      conversionOperationId: null,
      conversionId: null,
      diagnosticPackageLocation: null,
    });
  });

  it('should render nothing by default', () => {
    const view = render(<ConversionContent />);
    expect(view).toMatchSnapshot();
  });

  it('should render conversion content', () => {
    useConversionStore.setState({
      selectedStep: 1,
      conversionOperationLog: JSON.stringify({ foo: 'bar', baz: 'blee blue blah blueberry dee'}, null, 4),
      conversionOperationId: 'qqwer-asdf-zxcvzxcv-asdf-dgfh',
      conversionId: 17,
      diagnosticPackageLocation: 'http://conversion.content.com',
    });
    const view = render(<ConversionContent />);
    expect(view).toMatchSnapshot();
  });

  it('should render conversion content logs tab', () => {
    useConversionStore.setState({
      selectedStep: 1,
      conversionOperationLog: JSON.stringify({ foo: 'bar', baz: 'blee blue blah blueberry dee'}, null, 4),
      conversionOperationId: 'qqwer-asdf-zxcvzxcv-asdf-dgfh',
      conversionId: 17,
      diagnosticPackageLocation: 'http://conversion.content.com',
    });
    const view = render(<ConversionContent />);
    const logsTab = screen.getByText('logs');
    fireEvent.click(logsTab);
    expect(view).toMatchSnapshot();
  });
});