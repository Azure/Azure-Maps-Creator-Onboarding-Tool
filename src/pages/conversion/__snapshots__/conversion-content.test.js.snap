// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion content should render conversion content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.emotion-1 {
  margin-top: 1rem;
  margin-bottom: 1rem;
  font-size: 1rem;
}

.emotion-2 {
  font-weight: 600;
}

.emotion-3 {
  overflow: auto;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <span
            class="emotion-2"
          >
            ConversionId
          </span>
          : 
          17
        </div>
        <div
          class="emotion-3"
        >
          <h3>
            operation.log
          </h3>
          <div
            class="react-json-view"
            style="font-family: monospace; cursor: default; background-color: rgba(0, 0, 0, 0); position: relative;"
          >
            <div
              class="pretty-json-container object-container"
            >
              <div
                class="object-content"
              >
                <div
                  class="object-key-val"
                >
                  <span>
                    <span
                      style="display: inline-block; cursor: pointer;"
                    >
                      <div
                        class="icon-container"
                        style="display: inline-block; width: 17px;"
                      >
                        <span
                          class="expanded-icon"
                        >
                          <svg
                            fill="#586e75"
                            height="1em"
                            style="vertical-align: middle; color: rgb(88, 110, 117); height: 1em; width: 1em;"
                            viewBox="0 0 1792 1792"
                            width="1em"
                          >
                            <path
                              d="M1344 800v64q0 14-9 23t-23 9h-832q-14 0-23-9t-9-23v-64q0-14 9-23t23-9h832q14 0 23 9t9 23zm128 448v-832q0-66-47-113t-113-47h-832q-66 0-113 47t-47 113v832q0 66 47 113t113 47h832q66 0 113-47t47-113zm128-832v832q0 119-84.5 203.5t-203.5 84.5h-832q-119 0-203.5-84.5t-84.5-203.5v-832q0-119 84.5-203.5t203.5-84.5h832q119 0 203.5 84.5t84.5 203.5z"
                            />
                          </svg>
                        </span>
                      </div>
                      <span />
                      <span
                        style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                      >
                        {
                      </span>
                    </span>
                    <div
                      class="object-meta-data"
                      style="display: inline-block; padding: 0px 0px 0px 10px;"
                    />
                  </span>
                  <div
                    class="pushed-content object-container"
                  >
                    <div
                      class="object-content"
                      style="margin-left: 6px;"
                    >
                      <div
                        class="variable-row"
                        style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 10px;"
                      >
                        <span>
                          <span
                            class="object-key"
                            style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                          >
                            <span
                              style="vertical-align: top;"
                            >
                              "
                            </span>
                            <span
                              style="display: inline-block;"
                            >
                              foo
                            </span>
                            <span
                              style="vertical-align: top;"
                            >
                              "
                            </span>
                          </span>
                          <span
                            style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                          >
                            :
                          </span>
                        </span>
                        <div
                          class="variable-value"
                          style="display: inline-block; padding-right: 6px; position: relative; cursor: default;"
                        >
                          <div
                            style="display: inline-block; color: rgb(203, 75, 22);"
                          >
                            <span
                              class="string-value"
                              style="cursor: default;"
                            >
                              "
                              bar
                              "
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        class="variable-row"
                        style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 10px;"
                      >
                        <span>
                          <span
                            class="object-key"
                            style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                          >
                            <span
                              style="vertical-align: top;"
                            >
                              "
                            </span>
                            <span
                              style="display: inline-block;"
                            >
                              baz
                            </span>
                            <span
                              style="vertical-align: top;"
                            >
                              "
                            </span>
                          </span>
                          <span
                            style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                          >
                            :
                          </span>
                        </span>
                        <div
                          class="variable-value"
                          style="display: inline-block; padding-right: 6px; position: relative; cursor: default;"
                        >
                          <div
                            style="display: inline-block; color: rgb(203, 75, 22);"
                          >
                            <span
                              class="string-value"
                              style="cursor: default;"
                            >
                              "
                              blee blue blah blueberry dee
                              "
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <span
                    class="brace-row"
                  >
                    <span
                      style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
                    >
                      }
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.emotion-1 {
  margin-top: 1rem;
  margin-bottom: 1rem;
  font-size: 1rem;
}

.emotion-2 {
  font-weight: 600;
}

.emotion-3 {
  overflow: auto;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <span
          class="emotion-2"
        >
          ConversionId
        </span>
        : 
        17
      </div>
      <div
        class="emotion-3"
      >
        <h3>
          operation.log
        </h3>
        <div
          class="react-json-view"
          style="font-family: monospace; cursor: default; background-color: rgba(0, 0, 0, 0); position: relative;"
        >
          <div
            class="pretty-json-container object-container"
          >
            <div
              class="object-content"
            >
              <div
                class="object-key-val"
              >
                <span>
                  <span
                    style="display: inline-block; cursor: pointer;"
                  >
                    <div
                      class="icon-container"
                      style="display: inline-block; width: 17px;"
                    >
                      <span
                        class="expanded-icon"
                      >
                        <svg
                          fill="#586e75"
                          height="1em"
                          style="vertical-align: middle; color: rgb(88, 110, 117); height: 1em; width: 1em;"
                          viewBox="0 0 1792 1792"
                          width="1em"
                        >
                          <path
                            d="M1344 800v64q0 14-9 23t-23 9h-832q-14 0-23-9t-9-23v-64q0-14 9-23t23-9h832q14 0 23 9t9 23zm128 448v-832q0-66-47-113t-113-47h-832q-66 0-113 47t-47 113v832q0 66 47 113t113 47h832q66 0 113-47t47-113zm128-832v832q0 119-84.5 203.5t-203.5 84.5h-832q-119 0-203.5-84.5t-84.5-203.5v-832q0-119 84.5-203.5t203.5-84.5h832q119 0 203.5 84.5t84.5 203.5z"
                          />
                        </svg>
                      </span>
                    </div>
                    <span />
                    <span
                      style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                    >
                      {
                    </span>
                  </span>
                  <div
                    class="object-meta-data"
                    style="display: inline-block; padding: 0px 0px 0px 10px;"
                  />
                </span>
                <div
                  class="pushed-content object-container"
                >
                  <div
                    class="object-content"
                    style="margin-left: 6px;"
                  >
                    <div
                      class="variable-row"
                      style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 10px;"
                    >
                      <span>
                        <span
                          class="object-key"
                          style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                        >
                          <span
                            style="vertical-align: top;"
                          >
                            "
                          </span>
                          <span
                            style="display: inline-block;"
                          >
                            foo
                          </span>
                          <span
                            style="vertical-align: top;"
                          >
                            "
                          </span>
                        </span>
                        <span
                          style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                        >
                          :
                        </span>
                      </span>
                      <div
                        class="variable-value"
                        style="display: inline-block; padding-right: 6px; position: relative; cursor: default;"
                      >
                        <div
                          style="display: inline-block; color: rgb(203, 75, 22);"
                        >
                          <span
                            class="string-value"
                            style="cursor: default;"
                          >
                            "
                            bar
                            "
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="variable-row"
                      style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 10px;"
                    >
                      <span>
                        <span
                          class="object-key"
                          style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                        >
                          <span
                            style="vertical-align: top;"
                          >
                            "
                          </span>
                          <span
                            style="display: inline-block;"
                          >
                            baz
                          </span>
                          <span
                            style="vertical-align: top;"
                          >
                            "
                          </span>
                        </span>
                        <span
                          style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                        >
                          :
                        </span>
                      </span>
                      <div
                        class="variable-value"
                        style="display: inline-block; padding-right: 6px; position: relative; cursor: default;"
                      >
                        <div
                          style="display: inline-block; color: rgb(203, 75, 22);"
                        >
                          <span
                            class="string-value"
                            style="cursor: default;"
                          >
                            "
                            blee blue blah blueberry dee
                            "
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  class="brace-row"
                >
                  <span
                    style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
                  >
                    }
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`conversion content should render nothing by default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
